pkg/txn/storage/memorystorage/memorytable/table.go

```diff
@@ -0,0 +1,302 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package memorytable
+
+import (
+	"database/sql"
+	"sort"
+	"sync"
+	"sync/atomic"
+
+	"github.com/matrixorigin/matrixone/pkg/common/moerr"
+	"github.com/tidwall/btree"
+)
+
+type Table[
+	K Ordered[K],
+	V any,
+	R Row[K, V],
```

Comment:
Why do you need the Row?

Reply:
Hmm, I guess I got it.   So row is a generic of K, V.

Reply:
Yes. Row provides extra information such as indexes.